src/main/kotlin/org/rust/utils/formatterUtils.kt

```diff
@@ -0,0 +1,63 @@
+package org.rust.utils
+
+import com.intellij.psi.PsiComment
+import com.intellij.psi.PsiWhiteSpace
+import org.rust.lang.core.psi.*
+
+fun RsNamedElement.getFormattedParts(): Pair<String, String> {
```

Comment:
This part will cause conflict with #1020. There's now better implementation of signatures of various elements. And we also do similar work in [tree elements](https://github.com/intellij-rust/intellij-rust/tree/9f545171d9c40b5e223e3c85a4d6bcdcb2aac3d0/src/main/kotlin/org/rust/ide/structure) to show them in the project structure. So I'm totally for moving this functionality into a separate class and reuse it where needed.

So, I see three steps to achieve this:
1. Merge this PR.
2. I'll rebase my #1010 branch and replace this implementation of `getFormattedParts()` with the improved version.
3. I'll do the refactoring in tree structure to reuse this class and to move some ideas from there to `getFormattedParts()`. For instance, we effectively remove internal comments in tree structure, but leave them here so far, it should be fixed.

@matklad